test-examples.sh

```
@@ -0,0 +1,350 @@
+    
+    popd >/dev/null
+    
+    # Check for expected generated files
+    echo "  Checking generated files..."
+    local all_found=true
+    local missing_files=()
+    for expected_file in "${expected_files[@]}"; do
+        if [ -f "$example_dir/$expected_file" ]; then
+            echo -e "    ${GREEN}✓${NC} Found: $expected_file"
+            # Show file size to confirm it's not empty
+            local file_size=$(stat -f%z "$example_dir/$expected_file" 2>/dev/null || stat -c%s "$example_dir/$expected_file" 2>/dev/null || echo "unknown")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)